pkg/operator/cassandra/controller/util/util.go

```diff
@@ -210,3 +187,47 @@ func StatefulSetStatusesStale(c *cassandrav1alpha1.Cluster, statefulSetLister ap
 	}
 	return false, nil
 }
+
+// IsPVCsVolumeLost determines if a PVC's volume is on a node that
+// does not exist anymore.
+// To decide that, it examines the Volume's NodeSelector
+// and checks that each referenced node exists.
+func IsPVCsVolumeLost(pvc *corev1.PersistentVolumeClaim, kubeclient kubernetes.Interface) (bool, error) {
+
+	pvName := pvc.Spec.VolumeName
+	if pvName == "" {
+		return false, nil
+	}
+	pv, err := kubeclient.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Error trying to get PV %s : %s", pv.Name, err.Error())
+	}
+	if pv.Spec.NodeAffinity == nil {
+		return false, nil
+	}
+
+	// Check nodeAffinity for Nodes that do not exist
+	for _, term := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms {
+		for _, expr := range term.MatchExpressions {
+			if expr.Key == apis.LabelHostname && expr.Operator == corev1.NodeSelectorOpIn {
```

Comment:
i think there's a few supported ways to specify node selectors, does this approach to traversing them cover all the possibilities? is there any upstream helper code to iterate/traverse selectors instead of doing it ourselves?